hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JetTestSupport.java

```diff
@@ -146,6 +148,44 @@ public static void assertJobStatusEventually(Job job, JobStatus expected) {
         assertJobStatusEventually(job, expected, ASSERT_TRUE_EVENTUALLY_TIMEOUT);
     }
 
+    /**
+     * Asserts that a job status is eventually RUNNING. When it's running,
+     * checks that the execution ID is different from the given {@code
+     * ignoredExecutionId}, if not, tries again.
+     * <p>
+     * This is useful when checking that the job is running after a restart:
+     * <pre>{@code
+     *     job.restart();
+     *     // This is racy, we might see the previous execution running.
+     *     // Subsequent steps can fail because the job is restarting.
+     *     assertJobStatusEventually(job, RUNNING);
+     * }</pre>
+     *
+     * This method allows an equivalent code:
+     * <pre>{@code
+     *     long oldExecutionId = assertJobRunningEventually(instance, job, null);
+     *     // now we're sure the job is safe to restart - restart fails if the job isn't running
+     *     job.restart();
+     *     assertJobRunningEventually(instance, job, oldExecutionId);
+     *     // now we're sure that a new execution is running
+     * }</pre>
+     *
+     * @param ignoredExecutionId If job is running and has this execution ID,
+     *      wait longer. If null, no execution ID is ignored.
+     * @return the execution ID of the new execution
+     */
+    public static long assertJobRunningEventually(JetInstance instance, Job job, Long ignoredExecutionId) {
```

Comment:
Can we rename this method please? It does not assert whether job is running but whether job was restarted (and running). 

If this method is called independently on `job.restart();` it will never return. Should not be rather `job.restart();` called inside this method to have it more clear and robust?

Reply:
It does assert that the job is running, but it can **optionally** exclude certain execution ID. If the given execution ID is null, it ignores no execution ID. If it is not null, it's useful after a restart to ignore the previous execution.

What name do you suggest?

Reply:
You're right, I misread condition. What about the name like `assertJobRunningAfterRestartEventually` ? However since `ignoredExecutionId` can be `null` then name which I suggest is also bad. Maybe let's keep that name as it is.